MapboxNavigation/NavigationOptions.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+import MapboxCoreNavigation
+
+/*
+    Object representing available options for a Navigation Session.
+ */
+
+@objc(MBNavigationOptions)
+open class NavigationOptions: NSObject {
```

Comment:
If we define NavigationOptions in Objective-C as a C struct instead of a class, we get reference semantics in both languages without the mental overhead of allocation in Objective-C. (Note that [these days](https://developer.apple.com/videos/play/wwdc2018/409/) ARC supports object pointers in C structs.) On the other hand, C structs have a fixed size, so they aren’t as forward-compatible, especially as new options are added over time.

So it’s fine that we’ve defined it as a class here, but we should try to reduce the syntactic overhead in Objective-C, for example by adding a `+navigationOptions` initializer that takes no arguments. Additionally, history has shown that value classes like NavigationOptions quickly end up needing conformance to NSSecureCoding, object equality, and the like. These are all things we’d get for free with a struct, but we need to define them explicitly with a class (perhaps as tail work).

(The same considerations apply to RouteProgress et al., but reference semantics are much more useful in that case.)